src/maliput_malidrive/road_curve/spiral_ground_curve.cc

```diff
@@ -184,9 +184,89 @@ maliput::math::Vector2 SpiralGroundCurve::DoGDot(double p) const {
   return maliput::math::Vector2{std::cos(PowerOf(t, 2)), std::sin(PowerOf(t, 2))} * norm_;
 }
 
-double SpiralGroundCurve::DoGInverse(const maliput::math::Vector2&) const {
-  MALIPUT_THROW_MESSAGE("Unimplemented: SpiralGroundCurve::DoGInverse().");
-  return {};
+namespace {
+
+// @brief Find the best p parameter for @p curve that maps a point in the 2D INERTIAL-Frame that minimizes
+// the distance to @p point.
+// @details The function implements a binary search within a range trying to minimize the distance against
+// @p point. The range is split into equal @p partitions. The search can be early terminated if any of the
+// extents of the range at each iteration is within @p tolerance.
+// @param curve The SpiralGoundCurve to use.
+// @param point The point in the 2D INERTIAL-Frame to find a p parameter value for.
+// @param p0 The p parameter start range value. It must be non-negative.
+// @param p1 The p parameter end range value. It must be greater than @p p0 by GroundCurve::kEpsilon.
+// @param partitions The number of partitions of the range [@p p0; @p p1]. It must be greater or equal to 2.
+// @param tolerance The tolereance to early stop the search. It must be non-negative.
+// @return The p parameter value that minimizes the distance against @p point.
+double FindBestPParameter(const SpiralGroundCurve& curve, const maliput::math::Vector2& point, double p0, double p1,
```

Comment:
Not sure how it performs this binary search here however using the `maliput::math::KDTree` is possible here. The kdtree should be loaded at construct time(which might require some time when loading the map as you need to evaluate G at all the points) and in each gIneverse call the `nearest_point` method which will be quick